recipes/reflectcpp/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+from conan import ConanFile
+from conan.tools.files import get, copy, rmdir
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+
+import os
+
+required_conan_version = ">=1.54"
+
+
+class ReflectCppConan(ConanFile):
+    name = "reflectcpp"
+    description = "C++-20 library for fast serialization, deserialization and validation using reflection"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/getml/reflect-cpp"
+    topics = (
+        "reflection",
+        "serialization",
+        "memory",
+        "cbor",
+        "flatbuffers",
+        "json",
+        "msgpack",
+        "toml",
+        "xml",
+        "yaml",
+    )
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cbor": [True, False],
+        "with_flatbuffers": [True, False],
+        "with_msgpack": [True, False],
+        "with_toml": [True, False],
+        "with_xml": [True, False],
+        "with_yaml": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cbor": False,
+        "with_flatbuffers": False,
+        "with_msgpack": False,
+        "with_toml": False,
+        "with_xml": False,
+        "with_yaml": False,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        if not self.options.shared:
+            self.requires("ctre/3.9.0", transitive_headers=True)
+            self.requires("yyjson/0.8.0", transitive_headers=True)
```

Comment:
```suggestion
        if not self.options.shared:
            self.requires("ctre/3.9.0", transitive_headers=True)
            self.requires("yyjson/0.8.0", transitive_headers=True)
```

Why does adding those requirements only when static?

Reply:
If I set them to False, where would YYJSON come from? Like I said it can't come from the external libraries if they are linked statically but the library is linked dynamically.

Reply:
> If I set them to False, where would YYJSON come from?

It will be provided by Conan, using VirtualRunEnv generator:

https://docs.conan.io/2/reference/tools/env/virtualbuildenv.html

This generator is automatically executed. The test package is the example of it, it find all transitive dependencies by the configured environment variable (e.g LD_LIBRARY on Linux).